engine/access/rest/websockets/data_provider/blocks.go

```diff
@@ -0,0 +1,61 @@
+package data_provider
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine/access/state_stream"
+)
+
+type MockBlockProvider struct {
+	id               uuid.UUID
+	ch               chan<- interface{}
+	topic            string
+	logger           zerolog.Logger
+	ctx              context.Context
+	stopProviderFunc context.CancelFunc
+	streamApi        state_stream.API
+}
+
+func NewMockBlockProvider(
+	ctx context.Context,
+	ch chan<- interface{},
+	topic string,
+	logger zerolog.Logger,
+	streamApi state_stream.API,
+) *MockBlockProvider {
+	ctx, cancel := context.WithCancel(ctx)
+	return &MockBlockProvider{
+		id:               uuid.New(),
+		ch:               ch,
+		topic:            topic,
+		logger:           logger.With().Str("component", "block-provider").Logger(),
+		ctx:              ctx,
+		stopProviderFunc: cancel,
+		streamApi:        streamApi,
+	}
+}
+
+func (p *MockBlockProvider) Run() {
+	select {
+	case <-p.ctx.Done():
+		return
+	default:
+		p.ch <- "hello"
+		p.ch <- "world"
```

Comment:
one of the points from the design was to use a callback/interface to pass messages back to the controller. The implementation would then put the message on a channel. This allows the controller to stop accepting messages without risking a deadlock in one of the providers attempting to push a message.

Reply:
I use a channel here for clarity. It can be changed to taking callback/interface in the related task for `implementing response per second tracker`. Speaking of deadlock, I added this mock blocks provider only to make sure all parts work well together (here's a small test in this PR). It'll be removed in a task for the very first data provider